<!DOCTYPE html>
<!--[if lt IE 7]> <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]> <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]> <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js"> <!--<![endif]-->
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <title>Research  &middot; Genome Informatics</title>
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1">


<meta name="description" content="" />

<meta name="keywords" content="">


<meta property="og:title" content="Research  &middot; Genome Informatics ">
<meta property="og:site_name" content="Genome Informatics"/>
<meta property="og:url" content="https://rahmannlab.github.io/research/" />
<meta property="og:locale" content="en">


<meta property="og:type" content="article" />
<meta property="og:description" content=""/>
<meta property="og:article:published_time" content="2016-01-06T10:03:13Z" />
<meta property="og:article:modified_time" content="2016-01-06T10:03:13Z" />

  

  
<meta name="twitter:card" content="summary" />
<meta name="twitter:site" content="@GenomeInformaticsEssen" />
<meta name="twitter:creator" content="@GenomeInformaticsEssen" />
<meta name="twitter:title" content="Research" />
<meta name="twitter:description" content="" />
<meta name="twitter:url" content="https://rahmannlab.github.io/research/" />
<meta name="twitter:domain" content="https://rahmannlab.github.io/">
  

  
<script type="application/ld+json">
  {
    "@context": "http://schema.org",
    "@type": "Article",
    "headline": "Research",
    "author": {
      "@type": "Person",
      "name": "http://profiles.google.com/+?rel=author"
    },
    "datePublished": "2016-01-06",
    "description": "",
    "wordCount": 372
  }
</script>
  



<link rel="canonical" href="https://rahmannlab.github.io/research/" />
<link rel="apple-touch-icon-precomposed" sizes="144x144" href="https://rahmannlab.github.io/touch-icon-144-precomposed.png">
<link rel="icon" href="https://rahmannlab.github.io/favicon.png">
<meta name="generator" content="Hugo 0.46" />

  <!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.2/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->




    
    








    
    




    



<link rel="stylesheet" href="https://rahmannlab.github.io/css/bundle.min.0c0cd4da15bc6c22e952564f7827f0048eca5609ba9bc55b917bfd79229abfb3b1a735317dd232223cdfba3dfbba18336ca477b2fb98968f7fb7a2736b784fb7.css" integrity="sha512-DAzU2hW8bCLpUlZPeCfwBI7KVgm6m8VbkXv9eSKav7OxpzUxfdIyIjzfuj37uhgzbKR3svuYlo9/t6Jza3hPtw==">

</head>
<body data-ng-app="myapp" data-ng-controller="MyController" data-ng-mouseleave="MouseLeave($event)">
    <header id="main-header">
  <nav class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        
          
          <a class="navbar-brand-img" href="https://rahmannlab.github.io/">
            <img alt="Genome Informatics" src="logo.png">
            
          </a>
        </div>
        <div id="navbar" class="collapse navbar-collapse">
          <ul class="nav navbar-nav navbar-right">
            
            
            <li class="">
              
                <a href="https://rahmannlab.github.io" >
                  <i class='fa fa-road'></i>
                  About
                </a>
              
            </li>
            
            <li class="">
              
                <a href="https://rahmannlab.github.io/post/" >
                  
                  News
                </a>
              
            </li>
            
            <li class="">
              
                <a href="https://rahmannlab.github.io/lehre/" >
                  
                  Lehre
                </a>
              
            </li>
            
            <li class="">
              
                <a href="https://rahmannlab.github.io/research/" >
                  
                  Research
                </a>
              
            </li>
            
            <li class="">
              
                <a href="https://rahmannlab.github.io/people/" >
                  
                  People
                </a>
              
            </li>
            
            
              
          </ul>
        </div>
        
      </div>
    </nav>
  </header>





<div class="container">
  <div class="row">
    <div class="col-sm-9">


      <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12" >
<header class="container hat">
  <h1>Research
</h1>

</header>
</div>


      <div class="content">
  

<p>This page lists the current research topics of the group.</p>

<p>See also the <a href="https://rahmannlab.github.io/software">Software</a> page for publicly available Software.</p>

<h2 id="current-projects">Current Projects</h2>

<p><a href="https://rahmannlab.github.io/research/metatranscriptome-analysis.md"><strong>Metatranscriptome Analysis</strong></a></p>

<p>Daniela Beisser &amp; Sven Rahmann,</p>

<p>in collaboration with the Biodiversity group (Jens Boenigk), Essen</p>

<p><a href="https://rahmannlab.github.io/research/protein-hypernetworks"><strong>Protein Hypernetworks</strong></a></p>

<p>Bianca Stöcker, Johannes Köster &amp; Sven Rahmann,</p>

<p>in collaboration with Eli Zamir (MPI Stuttgart)</p>

<p><a href="https://rahmannlab.github.io/research/analysis-of-ddrad-seq-data-with-error-tolerant-hashing"><strong>Analysis of ddRAD sequencing data with error-tolerant hashing</strong></a></p>

<p>Henning Timm &amp; Sven Rahmann,</p>

<p>in collaboration with AG Aquatische Ökosystemforschung (Florian Leese), Essen</p>

<p><a href="https://rahmannlab.github.io/research/exome-analysis.md"><strong>Exome Analysis</strong></a></p>

<p>Christopher Schröder, Felix Mölder, Christoph Stahl &amp; Sven Rahmann,</p>

<p>in collaboration with the Institute of Human Genetics, Essen</p>

<p><a href="https://rahmannlab.github.io/research/whole-genome-methylation-analysis.md"><strong>Whole genome methylation analysis and epigenetic modifications</strong></a></p>

<p>Christopher Schröder &amp; Sven Rahmann,</p>

<p>in collaboration with the Institute of Human Genetics, Essen,</p>

<p>and with the DEEP consortium (German Epigenome Project)</p>

<p><a href="https://rahmannlab.github.io/research/library-complexity-and-duplication-rate-estimation.md"><strong>Library complexity and duplication rate estimation</strong></a></p>

<p>Christopher Schröder &amp; Sven Rahmann</p>

<p><a href="https://rahmannlab.github.io/research/osteosys.md"><strong>Osteosys: Osteoporose &#8211; eine Volkserkrankung im Kontext von kardiovaskulären Komplikationen und chronischer Inflammation: systemmedizinischer Ansatz zur personalisierten Therapie</strong></a></p>

<p>Felix Mölder &amp; Sven Rahmann</p>

<hr />

<h2 id="past-projects">Past Projects</h2>

<p><a href="https://rahmannlab.github.io/research/yeastscent.md"><strong>YeastScent</strong></a></p>

<p>Elias Kuthe &amp; Sven Rahmann,</p>

<p>in collaboration with Lars Blank, Christoph Halbfeld, Birgitta E. Ebert (RWTH Aachen), Sebastian Engell, Sven Wegerhoff (TU Dortmund), Jörg Ingo Baumbach, Ann-Kathrin Sippel (Reutlingen), Jessica Kuhlmann (B&amp;S Analytik GmbH, Dortmund), Michael Quantz, Erik Pollmann (Versuchsanstalt der Hefeindustrie e.V., Berlin)</p>

<p><a href="https://rahmannlab.github.io/research/dynatrait.md"><strong>Dynatrait</strong></a></p>

<p>Daniela Beisser &amp; Sven Rahmann,</p>

<p>in collaboration with Jens Boenigk and Christina Bock (Biodiversity, Essen), Bernd Sures (Aquatic Ecology, Essen) and Matthijs Vos (Theoretical and Applied Biodiversity, Bochum)</p>

<p><a href="http://genomeinformatics.uni-due.de/2016/07/25/simlord/"><strong>SimLoRD &#8211; simulation of long read data</strong></a></p>

<p>Bianca Stöcker, Johannes Köster &amp; Sven Rahmann</p>

<p>This is our simulator for PacBio sequence reads with an appropriate error model.</p>

<p><a href="https://rahmannlab.github.io/research/q-unique-sequences/"><strong>q-unique DNA Sequences</strong></a></p>

<p>Marianna D&#8217;Addario &amp; Sven Rahmann</p>

<p><a href="https://rahmannlab.github.io/research/ion-mobility-spectroscopy-ims-analysis-with-restricted-resources/"><strong>Analysis of ion mobility spectroscopy data with restricted resources</strong></a></p>

<p>Dominik Kopczynski, Marianna D&#8217;Addario &amp; Sven Rahmann,</p>

<p>in collaboration with Jörg Ingo Baumbach (Reutlingen) and B&amp;S Analytik GmbH</p>

<p>(This was a project in the Collaborative Research Center SFB 876.)</p>

<p><a href="https://rahmannlab.github.io/research/small-rna-sequencing"><strong>Small RNA and miRNA analysis</strong></a></p>

<p>Marcel Martin, Johannes Köster, Tobias Marschall, Sven Rahmann</p>

<p><a href="https://rahmannlab.github.io/research/probabilistic-arithmetic-automata-paas/"><strong>Probabilistic Arithmetic Automata (PAAs) and Applications</strong></a></p>

<p>Tobias Marschall &amp; Sven Rahmann</p>

<p><a href="https://rahmannlab.github.io/research/motif-discovery/"><strong>DNA Motif discovery</strong></a></p>

<p>Tobias Marschall &amp; Sven Rahmann</p>

<p><a href="https://rahmannlab.github.io/research/analysis-of-pattern-matching-algorithms/"><strong>Analysis of pattern matching algorithms</strong></a></p>

<p>Tobias Marschall &amp; Sven Rahmann</p>

<p><a href="https://rahmannlab.github.io/research/subsequence-combinatorics"><strong>Subsequence Combinatorics</strong></a></p>

<p>Sven Rahmann, with Cees Elzinga and Hui Wang</p>

<p><a href="https://rahmannlab.github.io/research/microarray-design-and-layout/"><strong>DNA Microarray Design and Layout</strong></a></p>

<p>Sérgio A. de Carvalho Jr &amp; Sven Rahmann</p>

</div>


      
    </div>
    
      <div class="col-xs-12 col-sm-12 col-md-3 col-lg-3">
        <div>
  

    <div class="section">
      <header><div class="title"><b>Latest Posts</b></div></header>
      <div class="content">
        <ul>
        
          <li>
          <a href="https://rahmannlab.github.io/2018/08/porting/">Hugo Port</a>
          </li>
        
          <li>
          <a href="https://rahmannlab.github.io/2016/04/what-is-hugo/">What is Hugo?</a>
          </li>
        
          <li>
          <a href="https://rahmannlab.github.io/2016/04/introduction/">Introduction</a>
          </li>
        
        </ul>
      </div>
    </div>

    
      
      
    
      
      
    

</div>

      </div>
    
  </div>
</div>
    
<footer class="footer hidden-print">
  <div class="container">
    <div class="row">
        <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12">
           <div class="pull-left">
  <a class="toplink" href="javascript:" id="return-to-top">back to top</a>
</div>
<div class="pull-right">

</div>

        </div>
        <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12 text-center">
              
    
<div class="container footline">
    <small>
  <img src=cropped-gi-banner.png style='height: 80%' alt='Logos'/>

</small>
</div>


    
<div class="container copyright">
    <small>
  &copy; 2018 Copyright Genome Informatics University of Duisburg Essen

  </small>
</div>



        </div>
    </div>
  </div>
</footer>

    

<script src="//s3.amazonaws.com/mailmunch/static/site.js" id="mailmunch-script" data-mailmunch-site-id="" async="async"></script>












<script>
var ENABLE_POPOVER = "", 
EXPIRE_COOKIE = "", 
SHOW_MODAL_TIMEOUT = "", 
MOUSE_LEAVE = "", 
MODAL_SIZE = "", 
POST_URL = "", 
SIGNUP_HEADER = "",
HEADER_IMAGE = "",
IMG_DESCRIPTION = "",
SIGNUP_TEXT = "",
INPUT_PLACEHOLDER = "",
SUBMIT_BUTTON = "",
SUCCESS_MESSAGE = "",
ERROR_MESSAGE = "",
OPTIN = "",
COOKIE_NAME = "",
CONTENTLANGUAGE = ""; 
</script>







<script  src="https://rahmannlab.github.io/js/bundle.min.a0761bf8598edda2bb2c0e348a3160c418f2bc30908f40740ee7f30dabbdeed27d82ee58e16400badd1ae84ce7be7ce10702c9be216abe5de1cb497de995e850.js" integrity="sha512-oHYb&#43;FmO3aK7LA40ijFgxBjyvDCQj0B0DufzDau97tJ9gu5Y4WQAut0a6EznvnzhBwLJviFqvl3hy0l96ZXoUA=="></script>



<script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js"></script>
<script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/config/TeX-AMS-MML_HTMLorMML.js"></script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Queue(function() {
    for(var all in MathJax.Hub.getAllJax()) {
        all.SourceElement().parentNode.className += ' has-jax';

    }
});
</script>





  </body>
</html>

