---
layout: person

role: scientist
# degree: MSc.
fullname: Johanna Schmitz
street: Campus, E2.1
room: '1.13'
city: 66123 Saarbrücken
tel: +49 (681) 302-70848
image: schmitz.jpg
# www:
where: uds
scholarlink: 'https://scholar.google.de/citations?user=tQxoNzAAAAAJ&hl=de&oi=ao'
---

I am a bioinformatician with a strong background in computer science and a focus on statistical modeling and data analysis. I am working in the Algorithmic Bioinformatics group, led by Sven Rahmann at Saarland University. My current topics are:
- Development of an efficient and accurate chromatin segmentation method based on epigenetic marks ([EpiSegMix](https://doi.org/10.1093/bioinformatics/btae178))
- Investigating the diversity or species richness of a large population

---

## Talks

|:---|:---|:---|:---|
| 2024 | [EpiSegMix: Discovering chromatin states using a flexible distribution hidden Markov model with duration modeling](/talks/dsb-2024-episegmix.pdf) | DSB | Montpellier |

## Posters

|:---|:---|:---|:---|
| 2024 | EpiSegMix: Discovering chromatin states using a flexible distribution hidden Markov model with duration modeling | ISMB | Montreal |
| 2024 | From the 1940s until today: An introduction to the unseen species problem | HIPS Symposium | Saarbrücken |
| 2024 | EpiSegMix: Discovering chromatin states using a flexible distribution hidden Markov model with duration modeling | Functional Epigenomics Conference | Saarbrücken |
| 2023 | Discovering chromatin states using a flexible distribution hidden Markov model with duration modeling | GCB | Hamburg |

---

## Tutorials

|:---|:---|:---|:---|
| 2024 | [ Just-in-Time Compiled Python for Bioinformatics Research](https://gcb2024.de/Workshops.html#WS5) | GCB | Bielefeld |
| 2024 | [Just-in-time compiled Python for bioinformatics research](https://www.iscb.org/ismb2024/programme-schedule/tutorials#ip2) | ISMB | Montréal |

---

## Summer Schools

|:---|:---|:---|
| 2024 | Computational Spatial Transcriptomics and Cell-Cell Communication with application in Biology and Medicine | Lipari |
| 2023 | Computational Single-Cell Analysis with application in Biology and Medicine | Lipari |

---

## Thesis Topics
 
|:---|:---|:---|
|Feb. 2023 | Master Thesis | Multivariate Hidden Markov Models with Flexible Distributions for Chromatin–State Discovery |
|Feb. 2021 | Bachelor Thesis | Identification of Key Transcriptional Regulators Using Single Cell Data |

---

## Teaching

### Courses

|:---|---:|:---|:---|:---|
| WS | 24/25 | BioStatsLab | Tutorials | Saarland University |
| SS | 2024 | Programming Course: Python | Tutorials | Saarland University | 
| WS | 23/24 | BioStatsLab | Tutorials | Saarland University |
| SS | 2023 | Elementare Effiziente Algorithmen für Stochastische Fragestellungen | Proseminar | Saarland University |
| WS | 22/23| Programming with Python | Tutorial | Saarland University |
| SS | 21| Bioinformatics 2 | Tutorial | Saarland University |
| WS | 20/21| Bioinformatics 1 | Tutorial | Saarland University |
| SS | 20 | Concurrent Programming | Tutorial | Saarland University |
| WS | 19/20| Programming 1 | Tutorial | Saarland University |