---
layout: person

role: scientist
# degree: MSc.
fullname: Jens Zentgraf
street: Campus, E2.1
room: '1.13'
city: 66123 Saarbrücken
tel: +49 (681) 302-70848
image: zentgraf.jpg
# www:
where: uds
scholarlink: 'https://scholar.google.com/citations?hl=en&user=_zJ2WDoAAAAJ'
---

I am a computer scientist with a focus on algorithm engineering and data analysis in bioinformatics. Right now I am working in the Algorithmic Bioinformatics group, led by Sven Rahmann at Saarland University. My current topics are:
- Hashing
  - Multi-Way Bucketed Cuckoo Hashing
  - Comparison with other Hashing methods
  - Parallelization using subtables
- Alignment free approaches using *k*-mers
  - [xengsort2](https://gitlab.com/genomeinformatics/xengsort)
  - [hackgap](https://gitlab.com/rahmannlab/hackgap)

---

## Talks

|:---|:---|:---|:---|
| 2025 | [Cleanifier: Removing human DNA contamination with a pangenomic gapped k-mer index](/talks/gcb2025.pdf) | GCB | Düsseldorf |
| 2025 | [Design of Worst-Case-Optimal Spaced Seeds](/talks/wabi2025.pdf) | WABI | online |
| 2025 | Blocked Bloom Filters with Choices | DSB | Pisa |
| 2024 | Efficient computing of strongly unique k-mers | GCB | Bielefeld |
| 2024 | Swiftly identifying strongly unique k-mers | WABI | London |
| 2023 | [Xengsort2: Ultrafast accurate xenograft sorting](/talks/gcb-2023-xengsort2.pdf) | GCB | Hamburg |
| 2022 | Fast gapped k-mer counting with subdivided multi-way bucketed Cuckoo hash tables | WABI | Potsdam |
| 2022 | Fast gapped k-mer counting with subdivided multi-way bucketed Cuckoo hash tables | DSB | Düsseldorf |
| 2020 | On an assignment problem for multi-way bucketed Cuckoo Hash tables on genomescale data | SIGOPT | Dortmund |
| 2020 | Cost-Optimal Assignment of Elements in Genome-Scale Multi-Way Bucketed Cuckoo Hash Tables | DSB | Rennes |
| 2020 | Cost-Optimal Assignment of Elements in Genome-Scale Multi-Way Bucketed Cuckoo Hash Tables | ALENEX | Salt Lake City |

---

## Posters

|:---|:---|:---|:---|
| 2024 |Xengsort2: Ultrafast accurate xenograft sorting | ISMB | Montreal |
| 2024 |Xengsort2: Ultrafast accurate xenograft sorting | HIPS Symposium | Saarbrücken |
| 2023 | *k*-mer counting combining super-*k*-mers and multi-way bucketed parallel Cuckoo hashing | CPM Summer school | Paris |
| 2022 | An efficient alignment-free method for finding genetic differences between pig races from individual whole genome sequencing data | Genome Informatics| Wellcome Genome Campus |
| 2020 | xengsort: Fast lightweight accurate xenograft sorting | HiTSeq | [online](https://www.youtube.com/watch?v=Ywy2BT0oIqc) |
| 2019 | Cost-optimal assignment of elements in genome-scale multi-way bucketed Cuckoo hash tables | GCB | Heidelberg |
| 2017 | PanGeA: Pan-Genome Annotation Indexing Annotated Human Genome Collections | GCB | Tübingen |

---
## Tutorials

|:---|:---|:---|:---|
| 2024 | [ Just-in-Time Compiled Python for Bioinformatics Research](https://gcb2024.de/Workshops.html#WS5) | GCB | Bielefeld |
| 2024 | [Just-in-time compiled Python for bioinformatics research](https://www.iscb.org/ismb2024/programme-schedule/tutorials#ip2) | ISMB | Montréal |
| 2021 | Modern hashing for alignment-free sequence analysis | GCB | online |
| 2020 | Fundamentals of alignment free sequence analysis: k-mer hashing | ACM-BCB | online |
| 2019 | Fundamentals of alignment free sequence analysis: k-mer hashing | GCB | Heidelberg |

## Participations

|:---|:---|:---|
|2025| Snakemake Hackathon | CERN |
|2024| DSB | Montpellier|
|2023| DSB | Delft |

---
---

## Teaching

### Courses

|:---|---:|:---|:---|:---|
| SS | 25 | Algorithms for Sequence Analysis | Tutorial | Saarland University |
| SS | 25 | Reproducible methods in metagenomics | (Pro-)Seminar | Saarland University |
| WS | 24/25 | Bioinformatik 1 | Tutorial | Saarland University |
| SS | 24 | Algorithms for Sequence Analysis | Tutorial | Saarland University |
| WS | 23/24 | Modern Hashing and Filtering Algorithms | (Pro-)Seminar | Saarland University |
| SS | 23 |  Algorithms for Sequence Analysis | Tutorials |  Saarland University |
| WS | 22/23 | Programming with Python | Tutorial | Saarland University |
| SS | 22 | Algorithms for Sequence Analysis | Tutorials |  Saarland University |
| WS | 21/22 | Statistics, Probability and Applications in Bioinformatics | Tutorials | Saarland University |
| SS | 21 | Effiziente Algorithmen | Tutorials |  TU Dortmund |
| WS | 20/21 | PG: GeCoCoo |  |  TU Dortmund |
| SS | 20 | PG: GeCoCoo |  |  TU Dortmund |
| WS | 19/20 | Einführung in die Programmierung | Tutorials | TU Dortmund |

### Bachelor Theses


|:---|:---|:---|
|2021| Tom Voellmer | Entdeckung bipartiter Motive in DNA-Sequenzen |
|2021| Alexander Korn | Effiziente Implementierung und Vergleich exotischer Pattern Matching Algorithmen |

### Master Theses

|:---|:---|:---|
|2023| Jens Petermann | Copy number variation detection using strongly unique *k*-mers and subdivided multi-way bucketed Cuckoo hash tables |
|2020| Marvin Böcker|Assembly of molecular graphs from induced subgraphs|

---
